src/entry_point.rs

```diff
@@ -47,7 +62,30 @@ unsafe impl GlobalAlloc for TockAllocator {
     }
 }
 
-/// Tock programs' entry point
+// _start and rust_start are the first two procedures executed when a Tock
+// application starts. _start is invoked directly by the Tock kernel; it
+// performs stack setup then calls rust_start. rust_start performs data
+// relocation and sets up the heap before calling the rustc-generated main.
+// rust_start and _start are tightly coupled: the order of rust_start's
+// parameters is designed to simplify _start's implementation.
+//
+// The memory layout set up by these methods is as follows:
+//
+//     +----------------+ <- app_heap_break
+//     | Heap           |
+//     +----------------| <- heap_bottom
+//     | Data (globals) |
```

Comment:
This picture does not (or at least: not yet) meet the memory layout we have in libtock-rs. Data currently lies in flash.